### ISSUE TITLE
Tested on Ubunu18.04 - Spanish Layout

### ISSUE THREAD

**Issue by pmarini** [JohnJohanssonChalmers/hp_wireless_keyboard#1]
Hi,

I've tested on a Ubuntu18.04 OS (more precisely, 18.04.2 LTS (Bionic Beaver)) with a Spanish layout keyboard.

```
$ lsusb
Bus 003 Device 003: ID 0461:4e90 Primax Electronics, Ltd
```

All works fine, except the numeric pad that shows strange behavior (it looks I have to press twice the entries to get the result)

Thanks for making this available and please let me know if you need any additional technical information.

Sorry if I open an issue, but I don't know other means to deliver my feedback!

**Comment by JohnJohanssonChalmers** [comment]
@pmarini Thank you for the feedback! Great to hear that it worked for you. So, what model of keyboard + mouse combo do you have? Is it the same as mine (HP Pavilion Wireless Keyboard and Mouse 800), except with Spanish Layout?

Your description of the num pad issue seems similar to what I have experienced, except that when I press a num pad key twice I typically get the correct character preceded by some other strange character.

No problem. I enabled "Issues" precisely so people could leave feedback just like you did.

**Comment by pmarini** [comment]
Hi, the model is: https://store.hp.com/us/en/pdp/hp-pavilion-wireless-keyboard-and-mouse-800
I guess is the same. 

In the meanwhile, let me mention that upgrading to Ubuntu18.10 made the numpad issue disappeared.  The numpad issue I had was exactly the same as yours. 

The only issue now is that after login (roughly less than 3 minutes) the screen turns black and only a combination of F<X> pads turns the desktop back to life. The issue does not occur again in the same session.

I would be happy to share more detailed information about my desktop environment (I guess it0's gnome) if only someone would tell me how.

**Comment by JohnJohanssonChalmers** [comment]
@pmarini Yes, we definitely have the same model (although mine is white and with Nordic layout).

Interesting that upgrading to Ubuntu 18.10 solved the num pad issue. I use Linux Mint 19.1, which is Based on Ubuntu 18.04, which is what you had before. Ubuntu 18.04 comes with Kernel 4.15, while Ubuntu 18.10 comes with Kernel 4.18. My guess is that this was addressed in Kernel 4.18.

Is it possible that the entire Primax issue was solved in Kernel 4.18.? Have you tried using the HP wireless combo on Ubuntu 18.10 without my driver fix?

I might have experienced something similar to your black screen issue, but not on a regular basis. Of course, I'm using Cinnamon, rather than Gnome, so maybe it's not the same issue. Does this happen every time you log in?

**Comment by pmarini** [comment]
Hi @JohnJohanssonChalmers,

I tried without your fix and it doesn't work. This means that your fix is still necessary for Ubuntu18.10

Regarding the black screen issue, it seems related in some way with the Gnome Desktop as switching to LXDE makes the issue disappeared.

As you ask me, yes the black screen issue used to happen at every login.

Do let me know if you want me to do some tests or post here some logs, I can easily switch beween graphical environments!
